src/foam/nanos/crunch/Capability.js

```diff
@@ -138,14 +139,68 @@ foam.CLASS({
       class: 'StringArray',
       documentation: `List of permissions granted by this capability`
     },
+    {
+      name: 'permissionsIntercepted',
+      class: 'StringArray',
+      documentation: `List of permissions intercepted by this capability`
+    },
     {
       name: 'daoKey',
-      class: 'String'
+      class: 'String',
+      documentation: `
+      daoKey.put() done in UserCapabilityJunctionDAO.
+      Uses contextDAOFindKey to find object to update/put.`
+    },
+    {
+      name: 'contextDAOFindKey',
+      class: 'String',
+      documentation: 'need to find things dynamically, thus have a string here to specify the object in context to look up.'
+    },
+    {
+      class: 'foam.mlang.predicate.PredicateProperty',
+      name: 'interceptIf',
+      javaFactory: `
+      return foam.mlang.MLang.TRUE;
+      `,
+      documentation: 'condition under which the permissions that may be intercepted by this capability will be intercepted.'
+    },
+    {
+      name: 'lastModified',
+      class: 'DateTime',
+      section: '_defaultSection',
+      createVisibility: 'HIDDEN',
+      updateVisibility: 'RO'
+    },
+    {
+      name: 'lastModifiedBy',
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      section: '_defaultSection',
+      createVisibility: 'HIDDEN',
+      updateVisibility: 'RO',
+      tableCellFormatter: function(value, obj) {
+        obj.userDAO.find(value).then(function(user) {
```

Comment:
Add a TODO to use projections to make more efficient.

Reply:
https://nanopay.atlassian.net/browse/NP-1373